wandb/sdk/wandb_run.py

```diff
@@ -3701,6 +3705,65 @@ def pb_to_dict(
                 logger.error("Error getting system metrics: %s", e)
         return {}
 
+    @property
+    @_run_decorator._attach
+    @_run_decorator._noop_on_finish()
+    def _metadata(self) -> Metadata | None:
```

Comment:
should we split the setter and getter (and also maybe cache the results, tho it is a handler call, so maybe it is not that expensive)